test/e2e/framework/pod/wait_test.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/onsi/ginkgo"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/output"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+// The line number of the following code is checked in TestFailureOutput below.
+// Be careful when moving it around or changing the import statements above.
+// Here are some intentionally blank lines that can be removed to compensate
+// for future additional import statements.
+//
+//
+//
+//
+//
+//
+//
+//
+//
+//
+// This must be line #47 (first line is #1).
+
+func runTests(t *testing.T, reporter ginkgo.Reporter) {
+	// This source code line will be part of the stack dump comparison.
+	ginkgo.RunSpecsWithDefaultAndCustomReporters(t, "Pod Suite", []ginkgo.Reporter{reporter})
+}
+
+var _ = ginkgo.Describe("pod", func() {
+	ginkgo.It("not found", func() {
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodRunningInNamespace(clientSet, "no-such-pod", "default", timeout), "wait for pod running")
+	})
+
+	ginkgo.It("not running", func() {
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodRunningInNamespace(clientSet, podName, podNamespace, timeout), "wait for pod running")
+	})
+})
+
+const (
+	podName      = "pending-pod"
+	podNamespace = "default"
+	timeout      = 5 * time.Second
+)
+
+var (
+	clientSet = fake.NewSimpleClientset(&v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: podName, Namespace: podNamespace}})
+)
+
+func TestFailureOutput(t *testing.T) {
+	// Sorted by name!
+	expected := output.SuiteResults{
+		output.TestResult{
+			Name: "[Top Level] pod not found",
+			Output: `INFO: Waiting up to 5s for pod "no-such-pod" in namespace "default" to be "running"
+INFO: Ignoring NotFound error while getting pod default/no-such-pod
+INFO: Ignoring NotFound error while getting pod default/no-such-pod
+INFO: Ignoring NotFound error while getting pod default/no-such-pod
+INFO: Ignoring NotFound error while getting pod default/no-such-pod
```

Comment:
Is the number of attempts stable enough (especially on over subscribed CI infrastructure) that this won't be flaky?

Reply:
I wasn't sure about that myself. It polls every two seconds, so yes, I guess on a busy system it might skip one or more of these polls. I'm reducing those lines to one now.